packages/bloc/lib/src/bloc.dart

```diff
@@ -1,9 +1,10 @@
 import 'dart:async';
 
-import 'package:bloc/bloc.dart';
 import 'package:meta/meta.dart';
 import 'package:rxdart/rxdart.dart';
 
+import '../bloc.dart';
```

Comment:
I thought we should prefer package imports for all. What rule are we breaking here?

Reply:
https://dart.dev/guides/language/effective-dart/usage#prefer-relative-paths-when-importing-libraries-within-your-own-packages-lib-directory

Reply:
Thanks, I'm still a bit confused because I was under the impression that the rule was only if you are referencing internal members (`./lib/src/*`) however in this case we are referencing public members of the library.

Reply:
As I understand it, it says that:

- you should never use `../` outside of lib and have `lib/` in your path
- you cannot reference from outside of `lib/` (`bin/`, `test/`, ...) with relative path
- in `lib/` prefer to use relative paths

it doesn't say anything about `lin/src/*` I believe. Maybe this sentence "Libraries inside lib can import other libraries inside lib (or in subdirectories of it)." is what confuses you?

This lint rule was implemented by https://github.com/dart-lang/linter/pull/1738. There are links to 2 issues where one of that, mine, references the specs from the Effective Dart. So it should be valid. :-)

Reply:
And it depends on you if you want to reference the public library file. I would personally opt for referencing only inside `lib/src/*`, as it probably makes the code a bit cleaner.

Reply:
@felangel should I keep it this way, or avoid importing a public library and therefore import other libraries directly? ;)

Reply:
I'm fine with keeping it as is